cwlgen/utils.py

```diff
@@ -1,16 +1,33 @@
 """
 Set of util functions and classes
 """
-
+import inspect
 
 class literal(str): pass
 
+_unparseable_types = [str, int, float, bool]
+
 
 def literal_presenter(dumper, data):
     return dumper.represent_scalar('tag:yaml.org,2002:str', data, style="|")
 
 
 class Serializable(object):
+    """
+    Serializable docstring
```

Comment:
Maybe describe this class here

Reply:
Added much better description:

> The Serializable class contains logic to automatically serialize a class based on
    its attributes. This behaviour can be overridden via the ``get_dict`` method on its
    subclasses with a call to super. Fields can be ignored by the base converter through
    the ``ignore_field_on_convert`` static attribute on your subclass.
>
> The parsing behaviour (beta) is similar, however it will attempt to set all attributes
    from the dictionary onto a newly initialised class. If your initialiser has required
    arguments, this converter will do its best to pull the id out of the dictionary to provide
    to your initializer (or pull it from the { $id: value } dictionary). Typing hints can be
    provided by the ``parse_types`` static attribute, and required attributes can be tagged
    the ``required_fields`` attribute.